src/main/java/net/pms/util/ImagesUtil.java

```diff
@@ -145,4 +155,98 @@ public static void parseImageByImaging(File file, DLNAMediaInfo media) throws Im
 			media.setCodecV(FormatConfiguration.TIFF);
 		}
 	}
+
+	/**
+	 * Add the format(container) name of the media to the generic icon image.
+	 *
+	 * @param image BufferdImage to be the label added
+	 * @param label the media container name to be added as an label
+	 * @param renderer the renderer configuration
+	 * 
+	 * @return the generic icon with the container label added and scaled in accordance with renderer setting
+	 */
+	public static InputStream addFormatLabelToImage(BufferedImage image, String label, RendererConfiguration renderer) throws IOException {
+		// copy image to not affect the original one
+		ColorModel cm = image.getColorModel();
+		BufferedImage img = new BufferedImage(cm, image.copyData(null), cm.isAlphaPremultiplied(), null);
+
+		Graphics2D g = img.createGraphics();
+		g.drawImage(img, 0, 0, null);
+		g.setColor(Color.WHITE);
+		g.setFont(new Font(Font.SANS_SERIF, Font.BOLD, 36));
+		g.drawString(label.toUpperCase(), 51, 62);
+		// XXX Is it necessary to use jpeg thumbnail-video with size 120x120 like it is used in the DLNAResource.getGenericThumbnailInputStream() method?
+		if (renderer != null) {
```

Comment:
If ```img.getWidth()``` or ```img.getHeight``` can ever be zero the next line will throw an ```ArithmeticException```. They should probably both be zero-checked in this if. If either is zero I guess scaling isn't important anyway.